application/Config/Services.php

```diff
@@ -30,13 +31,13 @@ class Services extends CoreServices
 //        return new \CodeIgniter\Example();
 //    }
 
-    public static function honeypot($getShared = true)
+    public static function honeypot(BaseConfig $config = null)
     {
-        if ($getShared)
-        {   
-            return self::getSharedInstance('honeypot');
```

Comment:
This line was removed, Because after the changes it keeps throwing error.
![screenshot_20180608_184010](https://user-images.githubusercontent.com/12835912/41176205-7ae5249c-6b57-11e8-9816-daa27a3c0ce2.png)

And i don't know the actual function of that line. If it's needed just suggest to me how to add it.

Reply:
Yeah - it is needed. That allows the instance to be cached and shared among all classes. The problem is that you need to pass the $config param into it. Basically, any params that get passed in the main function need to be passed into the `getSharedInstance` method:

```
return self::getSharedInstance('honeypot', $config);
```